packages/shared_preferences/shared_preferences/lib/tools/legacy_to_async_migration_tool.dart

```diff
@@ -0,0 +1,67 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:shared_preferences_platform_interface/types.dart';
+
+import '../shared_preferences.dart';
+
+/// A tool to migrate from the legacy SharedPreferences system to
```

Comment:
Similarly, I would say "Migrates preferences from [...]" so there's no potential confusion that this is a stand-alone tool that has to be packaged and run.